packages/components/src/internal/components/files/FileAttachmentForm.tsx

```diff
@@ -310,6 +297,32 @@ export class FileAttachmentForm extends React.Component<FileAttachmentFormProps,
             });
     }
 
+    processDataFile() {
+        const { previewGridProps, importFieldsFromJson } = this.props;
+        const { attachedFiles } = this.state;
+
+        // just take the first file, since we only support 1 file at this time
+        const file = attachedFiles.first();
+
+        // check if this usage has a set of formats which are supported for preview
+        if (previewGridProps.acceptedFormats) {
+            const fileCheck = fileMatchesAcceptedFormat(file.name, previewGridProps.acceptedFormats);
+            // if the file extension doesn't match the accepted preview formats, return without trying to get preview data
+            if (!fileCheck.get('isMatch')) {
+                return;
+            }
+        }
+
+        this.updatePreviewStatus('Uploading file...');
+
+        if (importFieldsFromJson) {
```

Comment:
Related to my other comments about this property, this doesn't seem quite right. If I upload a non-json file, I would want it to still go through the original code path of uploading the data file for file preview / infer fields. This code looks like now all file imports (i.e. xlsx and tsv files) would also call this importFieldsFromJson.

Reply:
Pulled apart 👍